agimus_controller/tests/test_ocp_croco_base.py

```diff
@@ -0,0 +1,223 @@
+
+            ### Creation of cost terms
+            ### Creation of cost terms
+            # State Regularization cost
+            x_residual = crocoddyl.ResidualModelState(
+                self._state,
+                np.concatenate((pin.neutral(self._rmodel), np.zeros(self._rmodel.nv))),
+            )
+            x_reg_cost = crocoddyl.CostModelResidual(self._state, x_residual)
+
+            # End effector frame cost
+            frame_placement_residual = crocoddyl.ResidualModelFramePlacement(
+                self._state,
+                self._rmodel.getFrameId("panda_hand_tcp"),
+                pin.SE3(np.eye(3), np.array([1.0, 1.0, 1.0])),
+            )
+
+            goal_tracking_cost = crocoddyl.CostModelResidual(
+                self._state, frame_placement_residual
+            )
+            terminal_cost_model.addCost("stateReg", x_reg_cost, 0.1)
+            terminal_cost_model.addCost("gripperPose", goal_tracking_cost, 50)
+
+            # Create Differential Action Model (DAM), i.e. continuous dynamics and cost functions
+            terminal_DAM = crocoddyl.DifferentialActionModelFreeFwdDynamics(
+                self._state,
+                self._actuation,
+                terminal_cost_model,
+            )
+
+            terminal_model = crocoddyl.IntegratedActionModelEuler(terminal_DAM, 0.0)
+            terminal_model.differential.armature = self._robot_model.armature
+
+            return terminal_model
+
+        def set_reference_horizon(self, horizon_size):
+            ### Not implemented in this OCP example.
+            return None
+
+    def setUp(self):
+        # Mock the RobotModelFactory and OCPParamsCrocoBase
+        self.robot_model_factory = RobotModelFactory()
+
+        robot = robex.load("panda")
+        robot_model = robot.model
+        collision_model = robot.collision_model
+        armature = np.full(robot_model.nq, 0.1)
+
+        self.robot_model_factory._rmodel = robot_model
+        self.robot_model_factory._complete_collision_model = collision_model
+        self.robot_model_factory.armature = armature
+
+        # Set mock parameters
+        self.ocp_params = OCPParamsBaseCroco(
+            dt=0.1, horizon_size=10, solver_iters=100, callbacks=True
+        )
+        self.state_reg = np.concatenate(
+            (pin.neutral(robot_model), np.zeros(robot_model.nv))
+        )
+        self.state_warmstart = [np.zeros(robot_model.nq + robot_model.nv)] * (
+            self.ocp_params.horizon_size - 1
+        )  # The first state is the current state
+        self.control_warmstart = [np.zeros(robot_model.nq)] * (
+            self.ocp_params.horizon_size - 1
+        )
+        # Create a concrete implementation of OCPBaseCroco
+        self.ocp = self.TestOCPCroco(self.robot_model_factory, self.ocp_params)
+        self.ocp.solve(self.state_reg, self.state_warmstart, self.control_warmstart)
+        # self.save_results()
```

Comment:
I think it's better to store results as a class value and then obtain them in each test case, rather than saing them to pickle

Reply:
I aggree we should have something like
```python
    def get_results_as_numpy(self):
         return np.array([
                self.ocp.ocp_results.states.tolist(),
                self.ocp.ocp_results.ricatti_gains.tolist(),
                self.ocp.ocp_results.feed_forward_terms.tolist(),
            ], dtype=object,  # Ensure the array is dtype=object before saving)

    def save_results(self, destination=pathlib.Path("ressources/simple_ocp_croco_results.npy")):
            np.save(str(destination), self.get_results_as_numpy())
```
The usage of this is:
```python
my_ocp = OCP()
my_ocp.solve()
my_ocp.save_results()
my_ocp.get_results_as_numpy()
```

Reply:
@Kotochleb is this what you meant?

Reply:
I think this would be better. Or even save a pickle with a dict with the numpy objects like so:
```python
results = {
   "states": ' self.ocp.ocp_results.states.tolist(),
   "ricatti_gains": ' self.ocp.ocp_results.ricatti_gains.tolist(),
   "feed_forward_terms": ' self.ocp.ocp_results.feed_forward_terms.tolist(),
}

with open("ressources/simple_ocp_croco_results.pkl", 'wb') as handle:
    pickle.dump(results, handle, protocol=pickle.HIGHEST_PROTOCOL)
```
Maybe this way if someone loads it, it will be more obvious what is inside of this object